### ISSUE TITLE
problem with linked lib when not using sudo (using ros_grant)

### ISSUE THREAD

**Issue by ugocupcic** [shadow-robot/ros_ethercat#6]
output is:

```
/opt/ros/hydro/lib/ros_ethercat_loop/ros_ethercat_loop: error while loading shared libraries: libcontroller_manager.so: cannot open shared object file: No such file or directory
```

**Comment by shadowmanos** [comment]
possible solution would be to reuse pr2-grant program.
we should test by getting the source of this program building it and if it succeeds replace the bash script by this program.
shadow-robot/sr-ros-interface-ethercat#36

**Comment by guihomework** [comment]
code is here :
https://code.ros.org/svn/pr2debs/trunk/packages/pr2/pr2-grant 
and compiles fine on ubuntu precise and trusty as long as libcap-dev is installed.
a make install is not sufficient for user to execute pr2grant, additionally "chmod u+s /usr/bin/pr2-grant" must be used to set the sticky bit. I guess this can be done in an installation package [EDIT] actually this is done in the svn repository already in the debian folder. One should maybe consider keeping this package completely but have it installed in indigo ?[/EDIT].

**Comment by ugocupcic** [comment]
@shadowmanos Are you happy with this:
- created a repo/package for ros_grant (https://github.com/shadow-robot/ros_grant)
- will release it applying patch to the bloom repo (http://answers.ros.org/question/191779/add-postinstall-rule-for-deb-package-creation/)
- added a readme for source install
- adding `ros_grant` to `launch-prefix` for our different launch files

This should be low maintenance (hopefully release it once per ros distro).

**Comment by toliver** [comment]
As far as I can see, using the xxx-grant will not solve the issue of not finding the linked lib. As the only thing grant does is allow you to set capabilities without being sudo.

**Comment by ugocupcic** [comment]
Comment from @shadowmanos (not sure why it's not appearing here)

Seems fine. Comments :
- We will also need a hydro-devel branch for our ronex drivers.
- If it's added in the launch files it should also be a runtime dependency in CMakeLists.txt and package.xml for every package that uses it in a launch file.
- If we are happy we should release this ASAP so that when we release a modified indigo-devel version of our hand bloom will find ros_grant in the servers.

**Comment by toliver** [comment]
I have found the cause (and a solution) to this issue.

When using capabilities (and also when using other things like setuid and setgid), the LD_LIBRARY_PATH variable is ignored, so the ROS linked libraries are not found.

The way this was solved for the pr2_ethercat executable is by using `rpath` when linking the executable or libraries used by it that will have to find other libraries. (See following link)

http://wiki.ros.org/hydro/Migration#pr2_robot.2BAC8-HydroMigration.Controllers_and_RPath
